File: php/Sintegra.php
<?php

/**
 * @author: Marlon Vismari
 * @date: 14/07/2015
 */
require_once 'autoload.php';

final class Sintegra extends Spider {

    // Colunas JSON
    public $NomeColunas = array(
            'CNPJ',
            'IncricaoEstadual',
            'RazaoSocial',
            'Logradouro',
            'Numero',
            'Complemento',
            'Bairro',
            'Municipio',
            'UF',
            'CEP',
            'Telefone',
            'AtividadeEconomica',
            'AtividadeEconomicaDataInicio',
            'SituacaoCadastralVigente',
            'DataDestaSituacaoCadastral',
            'RegimeDeApuracao',
            'EmitenteDeNFEDesde',
            'ObrigadaAEmitirNFEEm'
        );
    
   
    
    public function buscarDados($paramCNPJ) {
        // Página em HTML
        $PaginaEmHtml = $this->request('http://www.sintegra.es.gov.br/resultado.php', 'POST', 'http://www.sintegra.es.gov.br/index.php', array('num_cnpj' => $paramCNPJ, 'num_ie' => '', 'botao' => 'Consultar'));

        // Dados de saída
        $Informacoes = array();

        $Dom = new DOMDocument();
        $Dom->loadHTML($PaginaEmHtml);
        $DomXPath = new DOMXPath($Dom);
        $Nodes = $DomXPath->query("//td[@class='valor']");
        $iColuna = 0;
        foreach ($Nodes as $node) {
            if(isset($NomeColunas[$iColuna]))
                $Informacoes[$this->NomeColunas[$iColuna++]] = $node->nodeValue;
        }
        
        return json_encode(array('dados' => $Informacoes));
        
    }
}

File: ruby/Sintegra.rb
require './spider'
require 'nokogiri'
require 'json'

class Sintegra < Spider

  # Método construtor
  def initialize()
    super('http://www.sintegra.es.gov.br')
  end

  # Método para buscar os dados
  def buscarDados(paramCNPJ)
    @PaginaEmHtml = self.post('/resultado.php', {
                        'num_cnpj' => paramCNPJ, 'num_ie' => '', 'botao' => 'Consultar'
                    })

        parsed_data = Nokogiri::HTML.parse(@PaginaEmHtml.body)
        info = parsed_data.xpath("//td[@class='valor']")
       

        dados = {:dados => {
                    :cnpj => info[0].lstrip,
                    :inscricao_estadual => info[1].text.lstrip,
                    :razao_social => info[2].text.lstrip,
                    :logradouro => info[3].text.lstrip,
                    :numero => info[4].text.lstrip,
                    :complemento => info[5].text.lstrip,
                    :bairro => info[6].text.lstrip,
                    :municipio => info[7].text.lstrip,
                    :uf => info[8].text.lstrip,
                    :cep => info[9].text.lstrip,
                    :telefone => info[10].text.lstrip,
                    :atividade_economica => info[11].text.lstrip,
                    :atividade_economica_data_inicio => info[12].text.lstrip,
                    :situacao_cadastral_vigente => info[13].text.lstrip,
                    :data_desta_situacao_cadastral => info[14].text.lstrip,
                    :regime_de_apuracao => info[15].text.lstrip,
                    :emitente_de_nf_desde => info[16].text.lstrip,
                    :obrigado_a_emitir_nota_em => info[17].text.lstrip    
                }
            }
        
        puts JSON.pretty_generate(dados)
        end
end
